cmake/linker/ld/target.cmake

```
@@ -157,7 +157,7 @@ macro(toolchain_linker_finalize)
   string(REPLACE ";" " " zephyr_std_libs "${zephyr_std_libs}")
 
   set(link_libraries "<LINK_FLAGS> <OBJECTS> -o <TARGET> <LINK_LIBRARIES> ${zephyr_std_libs}")
```

it's probably better to remove the `<LINK_FLAGS>` at this line, because when we use c++ with exceptions we would like only the libraries and object files inside the `crtbegin` / `crtend` flags, and linker flags outside.

```suggestion
  set(link_libraries "<OBJECTS> -o <TARGET> <LINK_LIBRARIES> ${zephyr_std_libs}")
```